core/src/avm1/object/text_format_object.rs

```diff
@@ -1,67 +1,22 @@
-use crate::avm1::{Activation, Object, ScriptObject, TObject};
+use crate::avm1::object::NativeObject;
+use crate::avm1::{Activation, ScriptObject, TObject};
 use crate::html::TextFormat;
-use crate::impl_custom_object;
-use gc_arena::{Collect, GcCell, MutationContext};
-use std::cell::{Ref, RefMut};
-use std::fmt;
-
-#[derive(Clone, Copy, Collect)]
-#[collect(no_drop)]
-pub struct TextFormatObject<'gc>(GcCell<'gc, TextFormatData<'gc>>);
-
-#[derive(Collect)]
-#[collect(no_drop)]
-pub struct TextFormatData<'gc> {
-    /// The underlying script object.
-    base: ScriptObject<'gc>,
-
-    text_format: TextFormat,
-}
-
-impl fmt::Debug for TextFormatObject<'_> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let this = self.0.read();
-        f.debug_struct("TextFormatObject")
-            .field("text_format", &this.text_format)
-            .finish()
-    }
-}
-
-impl<'gc> TextFormatObject<'gc> {
-    pub fn empty_object(gc_context: MutationContext<'gc, '_>, proto: Option<Object<'gc>>) -> Self {
-        Self(GcCell::allocate(
-            gc_context,
-            TextFormatData {
-                base: ScriptObject::new(gc_context, proto),
-                text_format: TextFormat::default(),
-            },
-        ))
-    }
-
-    pub fn new(activation: &mut Activation<'_, 'gc, '_>, text_format: TextFormat) -> Self {
-        Self(GcCell::allocate(
+use gc_arena::GcCell;
+
+pub struct TextFormatObject;
+
+impl TextFormatObject {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new<'gc>(
+        activation: &mut Activation<'_, 'gc, '_>,
+        text_format: TextFormat,
+    ) -> ScriptObject<'gc> {
+        let proto = activation.context.avm1.prototypes().text_format;
+        let object = ScriptObject::new(activation.context.gc_context, Some(proto));
+        object.set_native(
```

Comment:
I suppose this is a first step to avoid many changes, but seems strange to have an unused `TextFormatObject` struct only to have a new method -- this could be a free function.

Perhaps longer term there is some trait like `avm1::ToObject` that we can `impl ToObject for TextFormat`, and similar for the other native types?

Reply:
For now I introduced a `new_text_format` free function in `avm1/globals/text_field.rs`. I agree we may need a more formal way for creating native objects, but I prefer to first port some more object types to see how it goes.